src/main/resources/com/mathworks/ci/MatlabBuilder/runMatlabTests.m

```diff
@@ -68,6 +85,84 @@
     end
 end
 
+% Produce Cobertura model coverage report (Not supported below R2018a) 
+if produceModelCoverage
+    if ~exist('sltest.plugins.ModelCoveragePlugin', 'class')
+        warning('MATLAB:testArtifact:cannotGenerateModelCoverageReport', ...
+                'Unable to generate Cobertura model coverage report. To generate the report, use a Simulink Coverage license with MATLAB R2018a or a newer release.');
+    else 
+        import sltest.plugins.ModelCoveragePlugin;
+        import matlab.unittest.plugins.codecoverage.CoberturaFormat;
+        
+        mkdirIfNeeded(resultsDir);
+        coverageFile = fullfile(resultsDir, 'coberturamodelcoverage.xml');
+        runner.addPlugin(ModelCoveragePlugin('Producing',CoberturaFormat(coverageFile)));
+    end
+end
+
+% Produce PDF test report (Not supported below R2016b)
+if producePDFReport && ~exportSTMResults
+    if ~testReportPluginPresent
+        issuePDFReportUnsupportedWarning();
+    else
+        import matlab.unittest.plugins.TestReportPlugin;
+        mkdirIfNeeded(resultsDir);
+        
+        if stmResultsPluginPresent
+            runner.addPlugin(TestManagerResultsPlugin);
+        end
+        runner.addPlugin(TestReportPlugin.producingPDF(getPDFFilePath(resultsDir)));
+    end
+end
+
+% Save Simulink Test Manager results in MLDATX format (Not supported below R2019a)
+if exportSTMResults && ~producePDFReport
+    if ~stmResultsPluginPresent || ~exportSTMResultsSupported
+        issueExportSTMResultsUnsupportedWarning();
+    else
+        mkdirIfNeeded(resultsDir);
+        runner.addPlugin(TestManagerResultsPlugin('ExportToFile', getMLDATXFilePath(resultsDir)));
+    end
+end
+
+% Note: We can remove the following piece of code once c4668443 gets ported to 19a branch.
+%
+% This check is to tackle the situation wherein user wants both pdf report and simulink test results.  
+% Basically, TestManagerResultsPlugin throws an error if we use two different instances of the
+% same plugin (g1898027). The geck is fixed but the changes are yet to be ported to 19a.
+if producePDFReport && exportSTMResults
+    try
+        import matlab.unittest.plugins.TestReportPlugin;
+        mkdirIfNeeded(resultsDir);
+
+        testReportPlugin = TestReportPlugin.producingPDF(getPDFFilePath(resultsDir));
+        stmResultsPlugin = TestManagerResultsPlugin('ExportToFile', getMLDATXFilePath(resultsDir));
+    catch exception
+        testReportPlugin = matlab.unittest.plugins.TestRunnerPlugin.empty;
+        stmResultsPlugin = matlab.unittest.plugins.TestRunnerPlugin.empty;
+        
+        if testReportPluginPresent && stmResultsPluginPresent && ...
+                exportSTMResultsSupported
+            throw(exception);
+        end
+        
+        if ~testReportPluginPresent
+            issuePDFReportUnsupportedWarning();
```

Comment:
In some cases MATLAB function with no parameter is called without () and in some cases with (). We should be consistent. Are there any MATLAB coding guidelines that apply here?

Reply:
I am not aware of any guidelines though but we don't need parantheses if there are no inputs to the function. Updated the code.